File: backend/src/main/java/com/nexus/backend/service/StudentService.java
package com.nexus.backend.service;

import com.nexus.backend.dto.request.UpdateAvailabilityRequest;
import com.nexus.backend.dto.request.UpdateProfileRequest;
import com.nexus.backend.dto.request.UpdateSocialLinksRequest;
import com.nexus.backend.dto.response.StudentProfileResponse;
import com.nexus.backend.dto.response.StudentResponse;

import java.util.List;

public interface StudentService {

    StudentProfileResponse getMyProfile();

    StudentProfileResponse updateProfile(
            UpdateProfileRequest request
    );

    StudentProfileResponse updateSocialLinks(
            UpdateSocialLinksRequest request
    );

    StudentProfileResponse updateAvailability(
            UpdateAvailabilityRequest request
    );

    // =====================================
    // Student Discovery
    // =====================================

    List<StudentResponse> getAllStudents();

    StudentResponse getStudentById(Long id);
}

File: frontend/src/services/studentService.js
import api from "./api";

// ==========================================
// GET MY PROFILE
// ==========================================
export const getMyProfile = async () => {
  const response = await api.get("/students/me");
  return response.data;
};

// ==========================================
// UPDATE PROFILE
// ==========================================
export const updateProfile = async (profileData) => {
  const response = await api.put("/students/profile", profileData);
  return response.data;
};

// ==========================================
// UPDATE SOCIAL LINKS
// ==========================================
export const updateSocialLinks = async (socialData) => {
  const response = await api.put("/students/social-links", socialData);
  return response.data;
};

// ==========================================
// UPDATE AVAILABILITY
// ==========================================
export const updateAvailability = async (availabilityData) => {
  const response = await api.put("/students/availability", availabilityData);
  return response.data;
};

// ==========================================
// CHANGE PASSWORD
// ==========================================
export const changePassword = async (passwordData) => {
  const response = await api.put(
      "/students/change-password",
      passwordData
  );
  return response.data;
};

// ==========================================
// STUDENT DISCOVERY
// ==========================================
export const getAllStudents = async () => {
  const response = await api.get("/students");
  return response.data;
};

export const getStudentById = async (id) => {
  const response = await api.get(`/students/${id}`);
  return response.data;
};